scripts/example.ts

```diff
@@ -0,0 +1,144 @@
+    },
+    create: {
+      description: 'Create the example from scratch (rather than degitting it)?',
+    },
+    forceDelete: {
+      description: 'Always delete an existing example, even if it has the same configuration?',
+    },
+    forceReuse: {
+      description: 'Always reusing an existing example, even if it has a different configuration?',
+    },
+    verdaccio: {
+      description: 'Use verdaccio rather than yarn linking stories?',
+    },
+    start: {
+      description: 'Start the example app?',
+      inverse: true,
+    },
+    build: {
+      description: 'Build the example app?',
+    },
+    watch: {
+      description: 'Start building used packages in watch mode as well as the example app?',
+    },
+    dryRun: {
+      description: "Don't execute commands, just list them?",
+    },
+  });
+}
+
+const steps = {
+  repro: {
+    command: 'repro',
+    description: 'Bootstrapping example',
+    icon: '👷',
+    hasArgument: true,
+    options: {
+      template: { values: frameworks },
+      e2e: {},
+    },
+  },
+  add: {
+    command: 'add',
+    description: 'Adding addon',
+    icon: '+',
+    hasArgument: true,
+    options: {},
+  },
+  build: {
+    command: 'build',
+    description: 'Building example',
+    icon: '🔨',
+    options: {},
+  },
+  dev: {
+    command: 'dev',
+    description: 'Starting example',
+    icon: '🖥 ',
+    options: {},
+  },
+};
+
+async function main() {
+  const optionValues = await getOptions();
+
+  const { framework, forceDelete, forceReuse, dryRun } = optionValues;
+  const cwd = path.join(examplesDir, framework as string);
+
+  const exists = await pathExists(cwd);
+  let shouldReuse = exists && forceReuse;
+  if (exists && !forceDelete && !forceReuse) {
+    ({ shouldReuse } = await prompts({
+      type: 'toggle',
+      message: `${path.relative(process.cwd(), cwd)} already exists, should we reuse it?`,
+      name: 'shouldReuse',
+      initial: true,
+      active: 'yes',
+      inactive: 'no',
+    }));
+  }
+
+  if (exists && !shouldReuse) await remove(cwd);
+
+  if (!shouldReuse) {
+    await executeCLIStep(steps.repro, {
+      argument: cwd,
+      optionValues: { template: framework },
+      cwd: examplesDir,
+      dryRun,
+    });
+
+    // TODO -- sb add <addon> doesn't actually work properly:
+    //   - installs in `deps` not `devDeps`
+    //   - does a `workspace:^` install (what does that mean?)
+    //   - doesn't add to `main.js`
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const addon of optionValues.addon as string[]) {
+      const addonName = `@storybook/addon-${addon}`;
+      // eslint-disable-next-line no-await-in-loop
+      await executeCLIStep(steps.add, { argument: addonName, cwd, dryRun });
+    }
+
+    // TODO copy stories
+  }
+
+  const { start } = optionValues;
+  if (start) {
+    await executeCLIStep(steps.dev, { cwd, dryRun });
```

Comment:
When I ran this, it was quite interesting: 
- The example was correctly bootstrapped
- It installed Storybook 6.5.9 as it used the current implementation of sb repro
- it ran:
```
🖥  Starting example
node /Users/yannbraga/open-source/storybook/code/lib/cli/bin/index.js dev
```

Because the CLI was built using SB 7.0 alpha, it failed as the project was bootstrapped with 6.5.9:

```
@storybook/cli v7.0.0-alpha.16

info => Loading presets
info => Loading presets
info => Loading presets
info => Loading presets
WARN   Failed to load preset: "@storybook/react/preset"
WARN   Failed to load preset: "@storybook/react/preset"
ERR!ERR! Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './preset' is not defined by "exports" in /Users/yannbraga/open-source/storybook/code/node_modules/@storybook/react/package.json
```

If I run `yarn storybook` in the example folder, it works fine.

We should probably do a few things:
1 - [maybe] Check if there is even a local CLI to use. People might have forgotten to run yarn bootstrap?
2 - Run yarn storybook instead of invoking the CLI directly, given the packages are locally linked